src/fixes/unordered_key.rs

```diff
@@ -279,6 +307,40 @@ mod tests {
         assert_lines(&lines, vec!["A=B", "B=C", "D=E"]);
     }
 
+    #[test]
+    fn key_order_substitution_variable_test() {
```

Comment:
Can you add more test cases?
For example: https://github.com/dotenv-linter/dotenv-linter/pull/401/files#diff-34a948562dd051763d9e109b86f2aad7d39f9e8ef8f072fcd674cebcfe001328R213

Reply:
Thank you for the comment! I have added many tests. One thing is unclear. On line 408 of `src/fixes/unordered_key.rs` I have a TODO comment because it's not clear to me whether a sort group that looks like:
```
A=1
B=2
C=$B$A
```
should result in
```
A=1
B=2
C=$B$A # Unordered, C uses A
```
or
```
A=1
B=2
C=$B$A # Unordered, C uses B
```

The test currently tests for the first case.

Let me know if you think I've missed anything!